### PR TITLE
Install create_policy as keylime_create_policy, annotate sources, and extend it

### PR DESCRIPTION
**Pull request by stefanberger**
This PR annotates the keylime/cmd/create_policy.py file (former scripts/create_policy.py) and makes it available as keylime_create_policy.

### CODE REVIEW

**Comment by stefanberger** [comment]
I had to rename create_policy to have a .py suffix otherwise mypy and pyright ignore it.

**Comment by aplanas** [comment]
I am working on https://github.com/keylime/keylime/pull/1317 (still in draft mode, I need to test it better), can we merge that before this one?

**Comment by stefanberger** [comment]
> I am working on #1317 (still in draft mode, I need to test it better), can we merge that before this one?

I would still need to adapt some references to create_policy in the user guide I think.

And then there's still this one here:

```
keylime/cmd/create_policy.py:284: error: Argument 1 to "ChainMap" has incompatible type "*List[Dict[str, List[str]]]"; expected "MutableMapping[str, str]"  [arg-type]
keylime/cmd/create_policy.py:376: error: Argument 1 to "ChainMap" has incompatible type "*List[Dict[str, List[Any]]]"; expected "MutableMapping[str, str]"  [arg-type]
```

File: keylime/cmd/create_policy.py
Comment (aplanas) [inline keylime/cmd/create_policy.py:507]: my apologies, I write the code without testing, as should be `policy[key].items()` (still not tested)

File: keylime/cmd/create_policy.py
Comment (aplanas) [inline keylime/cmd/create_policy.py:164]: do still need this and the rest of the changes?

Comment (stefanberger) [reply keylime/cmd/create_policy.py:164]: I know what you mean ... I would keep it because it makes repetitive code a lot shorter.

File: keylime/cmd/create_policy.py
Comment (stefanberger) [reply keylime/cmd/create_policy.py]: Ooops. Fixed.

Comment (stefanberger) [reply keylime/cmd/create_policy.py]: Support for excludes is still missing completely and once that's being added we can make sure that that list has also only unique values. But for that I'll hold off for another PR.

File: keylime/cmd/create_policy.py
Comment (aplanas) [reply keylime/cmd/create_policy.py:164]: To avoid the ugly repetition this can be used instead: `ima_buf_map.setdefault(path, []).append(checksum_hash)`

Comment (stefanberger) [reply keylime/cmd/create_policy.py:164]: Ha! Fixed.

File: keylime/cmd/create_policy.py
Comment (stefanberger) [reply keylime/cmd/create_policy.py]: ... rebase on latest master and added support for exclude list.

File: keylime/cmd/create_policy.py
Comment (aplanas) [inline keylime/cmd/create_policy.py:217]: why the change of name?

File: keylime/cmd/create_policy.py
Comment (aplanas) [inline keylime/cmd/create_policy.py:300]: (same)

File: keylime/signing.py
Comment (aplanas) [inline keylime/signing.py:17]: ideally should all be `PathLike_str`, as always the meaning will be for key_file to indicate a file

on the other side the API of this function is not very good, as it is used for the side effect of raising an exception

File: keylime/cmd/create_policy.py
Comment (stefanberger) [reply keylime/cmd/create_policy.py:217]: fixed.

File: keylime/cmd/create_policy.py
Comment (stefanberger) [reply keylime/cmd/create_policy.py:300]: Also fixed.

File: keylime/signing.py
Comment (stefanberger) [reply keylime/signing.py:17]: What do you mean? All callers should pass in a PathLike_str type?

**Review by aplanas — APPROVED** [review]

**Comment by stefanberger** [comment]
Setting this to draft status to merge this after PR #1331.

**Comment by stefanberger** [comment]
FYI, I have a PR for the test project to provide some test coverage for the keylime_create_policy tool: https://github.com/RedHat-SP-Security/keylime-tests/pull/341

**Review by maugustosilva — APPROVED** [review]

**Comment by maugustosilva** [comment]
IMO, LGTM

**Comment by stefanberger** [comment]
I have a pending PR on the test project to add coverage for this. I think it can be merged now.

**Pull request merged**
